src/NetEscapades.EnumGenerators/EquatableArray.cs

```diff
@@ -8,14 +8,20 @@ namespace NetEscapades.EnumGenerators;
 /// An immutable, equatable array. This is equivalent to <see cref="Array{T}"/> but with value equality support.
 /// </summary>
 /// <typeparam name="T">The type of values in the array.</typeparam>
-public readonly struct EquatableArray<T> : IEquatable<EquatableArray<T>>, IEnumerable<T>
+public readonly struct EquatableArray<T> : IEquatable<EquatableArray<T>>, IReadOnlyList<T>
```

Comment:
nit: I think we can revert the changes in this type now
```suggestion
public readonly struct EquatableArray<T> : IEquatable<EquatableArray<T>>, IEnumerable<T>
```

Reply:
It still helps the IDEs to know, that multiple enumarations are fine, and the old interface is still included